기능 추가

sungik-choi opened the pull request:
## 기능 추가

- 인덱스 페이지 인피니티 스크롤 추가
- 하위 컴포넌트, 커스텀 훅으로 관심사 분리
- 실제 마크다운 파일 생성
- gatsby-meta-config 파일로 설정 변수 분리
- 카테고리 정렬 : 글 많은 순으로

sungik-choi committed:
Feat : 폰트 안티앨리어싱 스타일 추가 및 폰트 굵기 변경

sungik-choi committed:
Feat : 마크다운 문서 업데이트 및 package.json 수정

sungik-choi committed:
Feat : gatsby-remark-copy-linked-files 패키지 설치

- gif 이미지 불러올 수 있도록

sungik-choi committed:
Chore : 사용하지 않는 패키지 제거

sungik-choi committed:
Feat : 그리드 파일 분리 & 인피니티 스크롤 추가

- 게시글이 많은 경우를 고려해 인피니티 스크롤을 추가
- 더 보기 버튼 & 히스토리 추가로 변경할 거 같음. 인피니티 스크롤은 UX에 좋지가 않다.. 푸터도 볼 수가 없고. 우선은 구현이 간단하므로 이방법으로 구현

sungik-choi committed:
Feat : 포스트 그리드 컴포넌트 제작 완료

- intersectionGrid의 obeserve 함수가 초기1회 무조건 일어난다. 이를 막기 위해 useState를 사용해서 초기 1회는 스킵하도록 만들었다.

sungik-choi committed:
Chore : 불필요한 주석 및 파일 삭제

sungik-choi committed:
Feat : 컴포넌트 디렉토리 이동

- Layout : layout 디렉토리 생성 후 이동
- ThemeContext : store 디렉토리 생성 후 이동
- postGrid 관련 컴포넌트 내부 디렉토리로 이동

sungik-choi committed:
Feat : gatsby-meta-config 파일 생성

- 메타데이터 관련 변수들 이동

sungik-choi committed:
Fix : 깜빡임 현상 해결

sungik-choi committed:
Feat : 카테고리 글 많은 순대로 상위 정렬 기능 추가

sungik-choi committed:
Docs : 문서 카테고리 변경

sungik-choi committed:
Feat : lang 메타데이터 별도 추가

sungik-choi committed:
Refactor : useSiteMetadata 디폴트 export로 변경 후 적용

sungik-choi committed:
Feat : navBar 컴포넌트에서 linkList 컴포넌트 분리

sungik-choi committed:
Feat : useScrollCenter 커스텀 훅 제작 후 분리

sungik-choi committed:
Feat : useInifiniteScroll 커스텀훅 제작 후 분리

sungik-choi committed:
Feat : 404 페이지 파티클 최소,최대값 설정

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**sungik-choi**

:tada: Your build was successful! [See the Deploy preview here.](https://sungikchoi-dev.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/0a1f03e2-4319-49d8-9562-417cea95901d/sites/47b582be-70ef-4f85-a003-16aaaf764a14/builds/ea482f7e-f288-4b11-beb6-7f1e37cc5d6f/details)

:clock1: Build time: **1m**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :green_heart: 99   |
| Accessibility |   :green_heart: 100   |
| Best Practices |   :green_heart: 100   |
| SEO |   :large_orange_diamond: 83   |

:link: [View full report](https://build-ea482f7e-f288-4b11-beb6-7f1e37cc5d6f.gtsb.io/reports/lighthouse/index.html)

sungik-choi committed:
Style : footer 텍스트 색상 수정

- 대비 높이기

sungik-choi committed:
Docs : 썸네일 추가

sungik-choi committed:
Feat : 이미지 쿼리문 수정

- thumbnail.png 로 모두 이름이 동일하므로 이미지를 제대로 가져오지 못하는 문제 발생
- thumbnail { childImageSharp { id } } 로 static 폴더에 생성될 이미지 id값에 접근할 수 있었음. 해당 고유 id값을 하도록 변경

sungik-choi committed:
Docs : README 업데이트
